lib/chef/provider/powershell_script.rb

```diff
@@ -130,7 +193,78 @@ def validate_script_syntax!
       # executed, otherwise 0 or 1 based on whether $? is set to true
       # (success, where we return 0) or false (where we return 1).
       def wrapper_script
-        <<~EOH
+        # The script looks very slightly different with inline powershell since
```

Comment:
Can we extract these two options into their own methods? Maybe something like `inline_powershell_wrapper...` and `shell_out_wrapper...`? The method is a bit long to start with, so the if/else is harder to read due to the length.

Reply:
absolutely can